### DISCUSSION TITLE
2FA verification reminder wants TOTP, allthough 'Security key' is set as preferred method

### DISCUSSION THREAD

**Discussion by hilmarm** [community/community#143271] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Having added a hardware key as my preferred method of 2FA, and disabling authentication app seems to be messed up. (See #139595 )

![Screenshot 2024-11-01 at 08-44-20 Verify your two-factor configuration](https://github.com/user-attachments/assets/db5bf72a-8835-4d88-a77a-95af76ed2a2e)

Now being logged in, I get a daily 'reminder', that I should check that my 2FA works, so I don't get locked out of my account.
However, the ONLY way of doing this, is using TOTP/Authentication app, even though I explicitly set my hardware key to be the preferred method to log in. So I assume this will now show up every day.

Looking for any solutions

**Comment by royw** [comment]
First, I agree asking for a TOTP authentication when the chosen 2FA method is hardware key is a serious bug.

I think I've worked around this issue... time will tell... ;-)

My guess is that I must have created a "verification code" on my iphone (iOS 18.0.1) as the new Password app shows that my github account has a "verification code".

Also my github account shows I somehow configure an Authenticator app.
- Settings, Password and authentication, Two-factor authentication, Two-factor methods, Authenticator app (Configured)

So I edited the Authenticator app.

Using the verification code from my iphone's Password app, the verification failed.

So deleted the verification code on the iphone, then scanned the QR code and added it to my github password.  

Entered the new verification code from the iphone into github test (verification code from the app) which then passed.  Finally just pressed the github Save button.

HTH
